pkg/services/ldap/ldap.go

```diff
@@ -362,7 +399,7 @@ func (auth *Auth) searchForUser(username string) (*UserInfo, error) {
 	}
 
 	if len(searchResult.Entries) == 0 {
-		return nil, ErrInvalidCredentials
+		return nil, ErrLDAPUserNotFound
```

Comment:
Not sure we would want to do this? From security stand point, this code would reveal that user does not exist by the type of the error, whereas previously (even before my refactoring), we would hide that info

Reply:
We should detect is user really not found and differ this from other errors, like LDAP connection problems or so. I return `ErrInvalidCredentials` error from the `Login()` function anyway (it seems, I forgot to do this in `SyncUser()`):
https://github.com/grafana/grafana/blob/disable-removed-ldap-users/pkg/services/ldap/ldap.go#L145-L148

I think it's possible to use `ErrInvalidCredentials` error type instead of `ErrLDAPUserNotFound` since `searchForUser()` doesn't validate the password. But `ErrLDAPUserNotFound` looks a little bit clearer than `ErrInvalidCredentials` in this case from my perspective.